proto/topsql_agent.proto

```diff
@@ -0,0 +1,27 @@
+syntax = "proto3";
+
+package resource_usage_agent;
+
+import "gogoproto/gogo.proto";
+import "rustproto.proto";
+
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.sizer_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+option (rustproto.lite_runtime_all) = true;
+
+option java_package = "org.tikv.kvproto";
+
+// ResourceUsageAgent is the persistent agent service for resource usage records
+service ResourceUsageAgent {
+  // CollectCPUTime is called periodically (e.g. per minute) to save the in-memory resource usage records
+  rpc CollectCPUTime(stream CPUTimeRequestTikv) returns (CPUTimeResponseTikv) {}
+}
+
+message CPUTimeRequestTikv {
+  repeated uint64 timestamp_list = 1;  // timestamp in second
+  repeated uint32 cpu_time_ms_list = 2;  // this can be greater than 1000 when counting concurrent running requests
+  bytes resource_tag = 3;  // the specific encoding is opaque to tikv-server
```

Comment:
```suggestion
  bytes resource_tag = 3;  // the tag of the resource, by which CPU time is grouped.
```

BTW since `resource_tag` is such important (as a Primary Key) I suggest to put it as the first field.